pkg/ebpf/driver_interface.go

```diff
@@ -0,0 +1,373 @@
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"sync"
+	"sync/atomic"
+	"unsafe"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"golang.org/x/sys/windows"
+)
+
+// HandleType expresses what of DriverHandle one is - implicitly implies if filters have been placed on the driver for the handle
+type HandleType string
+
+const (
+	driverFile = `\\.\ddfilter`
+
+	// FlowHandle is keyed to return 5-tuples from the driver that represents a flow. Used with: (#define FILTER_LAYER_TRANSPORT ((uint64_t) 1)
+	FlowHandle HandleType = "Flow"
+
+	// DataHandle is keyed to return full packets from the driver. Used with: #define FILTER_LAYER_IPPACKET ((uint64_t) 0)
+	DataHandle HandleType = "Data"
+
+	// StatsHandle has no filter set and is used to pull total stats from the driver
+	StatsHandle HandleType = "Stats"
+)
+
+var (
+	// Buffer holding datadog driver filterapi (ddfilterapi) signature to ensure consistency with driver.
+	ddAPIVersionBuf = makeDDAPIVersionBuffer(C.DD_FILTER_SIGNATURE)
+)
+
+// Creates a buffer that Driver will use to verify proper versions are communicating
+// We create a buffer because the system calls we make need a *byte which is not
+// possible with const value
+func makeDDAPIVersionBuffer(signature uint64) []byte {
+	buf := make([]byte, C.sizeof_uint64_t)
+	binary.LittleEndian.PutUint64(buf, signature)
+	return buf
+}
+
+// DriverInterface holds all necessary information for interacting with the windows driver
+type DriverInterface struct {
+	driverFlowHandle  *DriverHandle
+	driverStatsHandle *DriverHandle
+
+	path       string
+	totalFlows int64
+}
+
+// NewDriverInterface returns a DriverInterface struct for interacting with the driver
+func NewDriverInterface() (*DriverInterface, error) {
+	dc := &DriverInterface{
+		path: driverFile,
+	}
+
+	err := dc.SetupFlowHandle()
+	if err != nil {
+		return nil, fmt.Errorf("%s : %s", "error creating driver flow handle", err)
+	}
+
+	err = dc.SetupStatsHandle()
+	if err != nil {
+		return nil, fmt.Errorf("%s : %s", "error creating driver stats handle", err)
+	}
+
+	return dc, nil
+}
+
+func (di *DriverInterface) close() error {
+	err := windows.CloseHandle(di.driverFlowHandle.handle)
+	if err != nil {
+		log.Errorf("error closing flow file handle %v", err)
+	}
+	err = windows.CloseHandle(di.driverStatsHandle.handle)
+	if err != nil {
+		log.Errorf("error closing stat file handle %v", err)
+	}
+	return err
+}
+
+// SetupFlowHandle generates a windows Driver Handle, and creates a DriverHandle struct to pull flows from the driver
+// by setting the necessary filters
+func (di *DriverInterface) SetupFlowHandle() error {
+	h, err := di.GenerateDriverHandle()
+	if err != nil {
+		return err
+	}
+	dh, err := NewDriverHandle(h, FlowHandle)
+	if err != nil {
+		return err
+	}
+
+	// Prepare handle filters for each interface
+	err = dh.createDriverHandleFilters()
+	if err != nil {
+		return err
+	}
+
+	// Set handle filters
+	err = dh.setFilters()
+	if err != nil {
+		return err
```

Comment:
Do `createDriverHandleFilters`  and `setFilters` need to be separate functions?  They don't appear to be tested separately and struturing it this way makes it hard to see how the two functions interact   (sharing `dh.filters`).

Could we either:
a) consolidate into one function
b) have createDirverHandleFilters return a list of filters and setFilters take a list of filters instead of having the `filters` field